### ISSUE TITLE
failed to build docker binary within development container

### ISSUE THREAD

**Issue by ligc** [docker/docker#22688]
**Output of `docker version`:**

```
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.6.1
 Git commit:   20f81dd
 Built:        Wed, 20 Apr 2016 14:19:16 -0700
 OS/Arch:      linux/ppc64le

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.6.1
 Git commit:   20f81dd
 Built:        Wed, 20 Apr 2016 14:19:16 -0700
 OS/Arch:      linux/ppc64le
```

**Output of `docker info`:**

```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 33
Server Version: 1.10.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 30
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: host bridge null
Kernel Version: 4.4.0-21-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: ppc64le
CPUs: 8
Total Memory: 3.939 GiB
Name: c910f03c01p04
ID: 7YA7:GV5W:XHHS:AW3R:W3RZ:XUWE:5I4G:FL4U:GSKH:6RIL:CZMV:PQIS
Username: liguangcheng
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1.git clone https://github.com/docker/docker.git
2. cd docker
3. git checkout -b 1.10.3 v1.10.3
4. docker build -t docker-build -f Dockerfile.ppc64le .
5. docker run --privileged --rm -ti docker-build /bin/bash
6. #Inside the container, under directory /go/src/github.com/docker/docker
    hack/make.sh binary

**Describe the results you received:**
Failed to build the docker binary, with error "/usr/bin/ld: $WORK/github.com/mattn/libgo-sqlite3.a(sqlite3-binding.o): undefined reference to symbol 'pthread_mutex_trylock@@GLIBC_2.17'"

```
root@be7b8c139805:/go/src/github.com/docker/docker# hack/make.sh binary  ========> build 1.10.3 failed
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

bundles/1.10.3 already exists. Removing.

---> Making bundle: binary (in bundles/1.10.3/binary)
Building: bundles/1.10.3/binary/docker-1.10.3
go build: when using gccgo toolchain, please pass linker flags using -gccgoflags, not -ldflags
# github.com/docker/docker/docker
/usr/bin/ld: $WORK/github.com/mattn/libgo-sqlite3.a(sqlite3-binding.o): undefined reference to symbol 'pthread_mutex_trylock@@GLIBC_2.17'
//lib/powerpc64le-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
root@be7b8c139805:/go/src/github.com/docker/docker# 
root@be7b8c139805:/go/src/github.com/docker/docker# git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
root@be7b8c139805:/go/src/github.com/docker/docker# hack/make.sh binary ==========> build 1.12 also failed
# WARNING! I don't seem to be running in a Docker container.
# The result of this command might be an incorrect build, and will not be
# officially supported.
#
# Try this instead: make all
#

bundles/1.12.0-dev already exists. Removing.

---> Making bundle: binary (in bundles/1.12.0-dev/binary)
Building: bundles/1.12.0-dev/binary-client/docker-1.12.0-dev
go build: when using gccgo toolchain, please pass linker flags using -gccgoflags, not -ldflags
Created binary: bundles/1.12.0-dev/binary-client/docker-1.12.0-dev
Building: bundles/1.12.0-dev/binary-daemon/dockerd-1.12.0-dev
go build: when using gccgo toolchain, please pass linker flags using -gccgoflags, not -ldflags
# github.com/docker/docker/cmd/dockerd
/usr/bin/ld: $WORK/github.com/mattn/libgo-sqlite3.a(sqlite3-binding.o): undefined reference to symbol 'pthread_mutex_trylock@@GLIBC_2.17'
//lib/powerpc64le-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
root@be7b8c139805:/go/src/github.com/docker/docker# 

```

**Describe the results you expected:**
Expect to build the docker binary successfully.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by thaJeztah** [comment]
Can you build the current master? It's possible that the 1.10.3 tag doesn't build on ppc64le. We don't officially release packages for ppc64le, so patches that are needed to make it build, are usually not cherry-picked into the release branch / tag.

**Comment by ligc** [comment]
HI, 

Yes, I could build the master. as you mentioned that this is not formally supported to build ppc64le, I am fine to close this issue. 

BTW,  my purpose is to run 10K containers on a baremetal machine,  I tried 1.12 but ran into problem https://github.com/docker/docker/issues/22690, 1.10.3 has the problem https://github.com/docker/docker/pull/18362, so I wanted to update the SetMaxStack and recompile 1.10.3. For now, using 1.10.2 seems to be more promising, so I would rather not spending time on trying to recompile 1.10.3.

Thank you.

**Comment by thaJeztah** [comment]
1.10.3 actually fixed that issue through https://github.com/docker/docker/pull/18362 (iirc). And with the integration of containerd may actually scale better. Have you tried setting TasksMax as I mentioned in https://github.com/docker/docker/issues/22690?

**Comment by ligc** [comment]
I am not sure for the fix with 1.10.3, I actually ran into the issue after created 2000+ containers with 1.10.3.

Yes, I changed the TasksMax  as you suggested in #22690, it is going much further(3500+ now), still in the progress.
